it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyControllerISpec.scala

```diff
@@ -271,6 +272,46 @@ trait CheckAgencyControllerISpec extends BaseISpec with SessionDataMissingSpec {
       }
       e.getMessage should include(validUtr.value)
     }
+
+    "showAlreadySubscribed, when fully subscribed" in {
+      withMatchingUtrAndPostcode(validUtr, validPostcode, isSubscribedToAgentServices = true, isSubscribedToETMP = true)
+      implicit val request = authenticatedAs(subscribingAgentEnrolledForNonMTD)
+        .withFormUrlEncodedBody("utr" -> validUtr.value, "postcode" -> validPostcode)
+      val result = await(controller.checkAgencyStatus(Some(CheckAgencyController.validBusinessTypes.head))(request))
+      redirectLocation(result) shouldBe Some(routes.CheckAgencyController.showAlreadySubscribed().url)
+    }
+
+    "subscriptionComplete for partiallySubscribed User" in {
+      withMatchingUtrAndPostcode(validUtr, validPostcode, isSubscribedToAgentServices = false, isSubscribedToETMP = true)
+      AgentSubscriptionStub.partialSubscriptionWillSucceed(CompletePartialSubscriptionBody(validUtr,
+        knownFacts = SubscriptionRequestKnownFacts(validPostcode)))
+
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments)
+        .withFormUrlEncodedBody("utr" -> validUtr.value, "postcode" -> validPostcode)
+      val result = await(controller.checkAgencyStatus(Some(CheckAgencyController.validBusinessTypes.head))(request))
+      redirectLocation(result) shouldBe Some(routes.SubscriptionController.showSubscriptionComplete().url)
+    }
+
+    "goGetCleanCreds, creds with enrolment/s are not allowed when partiallySubscribed User" in {
```

Comment:
`goGetCleanCreds` - is that `showHasOtherEnrolments`?

Reply:
yes I will rename to showHasOtherEnrolments